tests/network/cluster_addons_operator/test_network_addons_operator.py

```diff
@@ -149,21 +159,34 @@ def check_components(network_addons_config_scope_session):
 
 @pytest.fixture(scope="module")
 def net_add_op_bridge_device(worker_node1):
-    with utilities.network.network_device(
-        interface_type=LINUX_BRIDGE,
-        nncp_name="test-network-operator",
-        interface_name="net-add-br",
+    desired_state = DesiredState(
+        interfaces=[
+            Interface(
+                name="net-add-br",
+                state=NodeNetworkConfigurationPolicy.Interface.State.UP,
+                type=LINUX_BRIDGE,
+                ipv4=IPv4(dhcp=False),
+                ipv6=IPv6(dhcp=False),
+                bridge=Bridge(BridgeOptions(STP(enabled=False))),
+            )
+        ]
+    )
+    with NodeNetworkConfigurationPolicy(
+        name="test-network-operator",
+        desired_state=desired_state,
         node_selector=get_node_selector_dict(node_selector=worker_node1.hostname),
     ) as br_dev:
+        br_dev.wait_for_status_success()
         yield br_dev
 
 
 @pytest.fixture(scope="module")
 def net_add_op_br1test_nad(namespace, net_add_op_bridge_device):
+    interface_name = net_add_op_bridge_device.instance["spec"]["desiredState"]["interfaces"][0]["name"]
```

Comment:
1. You do not need to re-read the nncp object. It is returned after it has been successfully deployed.
2. You should use a getter (as a property) to fetch the spec (`desired_state`) and access the relevant field.

E.g.
```python
interface_name = nncp_with_bridge.desired_state.interfaces[0].name
```